### PR TITLE
Feature/cax portal@cplp 569 table component

### PR DESCRIPTION
**Pull request by fatihayyildiz**
- style of data table component created
- Storybook story implemented for basic table component
- MUI Datatable library added to package.json
- Data map functions created and moved under utils folder -> Temporary maps function
- Table component props exported  and default props defined
- Actions file created for async thunk actions

### CODE REVIEW

File: portal/code/cx-portal-shared-components/src/components/basic/Table/components/RoleChip.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import './RoleChip.scss'
+// Temporary component to visualise different table cell
+const RoleChip = ({
+                         role
+                       }: any) => {
```
Comment (oyo) [inline portal/code/cx-portal-shared-components/src/components/basic/Table/components/RoleChip.tsx]: I suggest we all run a `yarn pretty` or before checking in the code so we have a uniform code style

File: portal/code/cx-portal/src/utils/dataMapper.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import {
```
Comment (oyo) [inline portal/code/cx-portal/src/utils/dataMapper.ts:1]: for this kind of logic we should provide Unit tests - though it's a very simple one here

File: portal/code/cx-portal/src/state/features/partnerNetwork/partnerNetworkActions.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  'partnerNetwork/getOneBusinessPartner', async (bpn: string) => {
+    try {
+      // Call axios instance to get values
+      if (store.getState().user?.token) {
+        const partnerNetworkApi = api.PartnerNetworkApi.getInstance(store.getState().user?.token)
+        return await partnerNetworkApi.getBusinessPartnerByBpn(bpn)
+      }
+    } catch (error: any) {
+      console.error('api call error:', error)
+      throw Error('getOneBusinessPartner api call error')
```
Comment (oyo) [inline portal/code/cx-portal/src/state/features/partnerNetwork/partnerNetworkActions.ts:32]: why do you throw this error and not the one in the previous thunk?

**Review by oyo — APPROVED** [review]
generally approved, but check the comments

File: portal/code/cx-portal-shared-components/src/components/basic/Table/components/RoleChip.tsx (hunk 1 above)
Comment (fatihayyildiz) [reply portal/code/cx-portal-shared-components/src/components/basic/Table/components/RoleChip.tsx]: done

File: portal/code/cx-portal/src/state/features/partnerNetwork/partnerNetworkActions.ts (hunk 3 above)
Comment (fatihayyildiz) [reply portal/code/cx-portal/src/state/features/partnerNetwork/partnerNetworkActions.ts:32]: Agree I just forgot to put in other action :)

File: portal/code/cx-portal/src/utils/dataMapper.ts (hunk 2 above)
Comment (fatihayyildiz) [reply portal/code/cx-portal/src/utils/dataMapper.ts:1]: Would be nice, I noted it and I will provide in another PR

**Comment by fatihayyildiz** [comment]
I updated PR and beside that unit test part, all resolved. I will merge this PR to main branch

**Pull request merged**